backend/src/config.ts

```diff
@@ -30,16 +75,61 @@ export class Config {
             DFM_WEBSOCKET_PORT_TESTING: tcpPortValidator({ default: 13200 }),
             DFM_HTTP_PORT: tcpPortValidator({ default: 3201 }),
             DFM_HTTP_PORT_TESTING: tcpPortValidator({ default: 13201 }),
+            DFM_DB_USER: str(),
+            DFM_DB_USER_TESTING: str({ default: undefined }),
+            DFM_DB_PASSWORD: str(),
+            DFM_DB_PASSWORD_TESTING: str({ default: undefined }),
+            DFM_DB_NAME: str(),
+            DFM_DB_NAME_TESTING: str({ default: undefined }),
+            DFM_DB_LOG: bool({ default: false }),
+            DFM_DB_LOG_TESTING: bool({ default: undefined }),
+            DFM_DB_HOST: str({ default: 'localhost' }),
+            DFM_DB_HOST_TESTING: str({ default: 'localhost' }),
+            DFM_DB_PORT: tcpPortValidator({ default: 5432 }),
+            DFM_DB_PORT_TESTING: tcpPortValidator({ default: 5432 }),
         });
     }
 
-    public static initialize(testing: boolean = false) {
+    private static isInitialized = false;
+
+    private static throwIfNotInitialized() {
+        if (!this.isInitialized)
+            throw new Error('Config was used uninitialized');
```

Comment:
Maybe instead of `Config was used unitialized`, is there a way to dynamically use something like `dbName was used unitialized`? (using the name)

Reply:
As far as I understood it, this error should only be thrown if Config.initialize() hasn't been called before accessing any property.
It doesn't matter for this contract what exactly was accessed.
For debugging the Error object creates a stack trace on creation.

Reply:
@Dassderdie is completely right with this. I don't see a reason to make this more complicated.